### PR TITLE
Add telemetry dump of Options

### PR DESCRIPTION
**Pull request by ryzngard**
This adds telemetry for when a solution is closed to dump all current options. 

## Navigating the Code 

* `OptionsLogger.ReportTelemetry` is the method handling what telemetry is actually reported
* `SolutionEventMonitor.SolutionClosed` was added as an event
* `ReportSessionWideTelemetry` was added before `DisposeVisualStudioServices` to make sure telemetry services weren't disposed

### CODE REVIEW

**Comment by ryzngard** [comment]
# Example

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">



<body link="#0563C1" vlink="#954F72">



Property | Value
-- | --
vs.ide.vbcs.optionslogger.logoptions.adornment_editorcomponentonoffoptions | TRUE
vs.ide.vbcs.optionslogger.logoptions.allowblanklineaftercoloninconstructorinitializer_csharpcodestyleoptions | \<CodeStyleOption SerializationVersion="1"   Type="Boolean" Value="true"   DiagnosticSeverity="Hidden" />
vs.ide.vbcs.optionslogger.logoptions.allowblanklinesbetweenconsecutivebraces_csharpcodestyleoptions | \<CodeStyleOption SerializationVersion="1"   Type="Boolean" Value="true"   DiagnosticSeverity="Hidden" />
vs.ide.vbcs.optionslogger.logoptions.allowembeddedstatementsonsameline_csharpcodestyleoptions | \<CodeStyleOption SerializationVersion="1"   Type="Boolean" Value="true"   DiagnosticSeverity="Hidden" />
vs.ide.vbcs.optionslogger.logoptions.asynchronous_suggestionsoptions | null
vs.ide.vbcs.optionslogger.logoptions.asynchronousquickactionsdisablefeatureflag_suggestionsoptions | FALSE
vs.ide.vbcs.optionslogger.logoptions.automaticallycompletestatementonsemicolon_featureonoffoptions | TRUE
vs.ide.vbcs.optionslogger.logoptions.automaticlineender_internalfeatureonoffoptions | TRUE
vs.ide.vbcs.optionslogger.logoptions.bracematching_internalfeatureonoffoptions | TRUE
vs.ide.vbcs.optionslogger.logoptions.classification_internalfeatureonoffoptions | TRUE
vs.ide.vbcs.optionslogger.logoptions.coderefactorings_editorcomponentonoffoptions | TRUE
vs.ide.vbcs.optionslogger.logoptions.disablecrashingextensions_extensionmanageroptions | TRUE
vs.ide.vbcs.optionslogger.logoptions.displayallhintswhilepressingaltf1_inlinehintsoptions | TRUE
vs.ide.vbcs.optionslogger.logoptions.enabledfeatureflag_keybindingresetoptions | FALSE
vs.ide.vbcs.optionslogger.logoptions.enablelspintellisensefeatureflag_xamloptions | FALSE
vs.ide.vbcs.optionslogger.logoptions.etwloggerkey_loggeroptions | TRUE
vs.ide.vbcs.optionslogger.logoptions.eventhookup_internalfeatureonoffoptions | TRUE
vs.ide.vbcs.optionslogger.logoptions.fileheadertemplate_codestyleoptions |  
vs.ide.vbcs.optionslogger.logoptions.formatonsave_internalfeatureonoffoptions | TRUE
vs.ide.vbcs.optionslogger.logoptions.fullsolutionanalysismemorymonitor_internalfeatureonoffoptions | TRUE
vs.ide.vbcs.optionslogger.logoptions.implicitobjectcreationwhentypeisapparent_csharpcodestyleoptions | \<CodeStyleOption SerializationVersion="1"   Type="Boolean" Value="true"   DiagnosticSeverity="Info" />
vs.ide.vbcs.optionslogger.logoptions.indentblock_csharpformattingoptions | TRUE
vs.ide.vbcs.optionslogger.logoptions.indentbraces_csharpformattingoptions | FALSE
vs.ide.vbcs.optionslogger.logoptions.indentswitchcasesection_csharpformattingoptions | TRUE
vs.ide.vbcs.optionslogger.logoptions.indentswitchcasesectionwhenblock_csharpformattingoptions | TRUE
vs.ide.vbcs.optionslogger.logoptions.indentswitchsection_csharpformattingoptions | TRUE
vs.ide.vbcs.optionslogger.logoptions.inheritancemargincombinedwithindicatormargin_featureonoffoptions | FALSE
vs.ide.vbcs.optionslogger.logoptions.insertfinalnewline_formattingoptions | FALSE
vs.ide.vbcs.optionslogger.logoptions.labelpositioning_csharpformattingoptions | OneLess
vs.ide.vbcs.optionslogger.logoptions.logtelemetryforbackgroundanalyzerexecution_diagnosticoptions | FALSE
vs.ide.vbcs.optionslogger.logoptions.lspeditorfeatureflag_lspoptions | FALSE
vs.ide.vbcs.optionslogger.logoptions.lsppulldiagnosticsfeatureflag_diagnosticoptions | FALSE
vs.ide.vbcs.optionslogger.logoptions.maxcompletionlistsize_lspoptions | 1000
vs.ide.vbcs.optionslogger.logoptions.namespacedeclarations_csharpcodestyleoptions | \<CodeStyleOption SerializationVersion="1"   Type="Boolean" Value="false"   DiagnosticSeverity="Info" />
vs.ide.vbcs.optionslogger.logoptions.needsreset_keybindingresetoptions | FALSE
vs.ide.vbcs.optionslogger.logoptions.nevershowagain_keybindingresetoptions | FALSE
vs.ide.vbcs.optionslogger.logoptions.newlineforcatch_csharpformattingoptions | TRUE
vs.ide.vbcs.optionslogger.logoptions.newlineforclausesinquery_csharpformattingoptions | TRUE
vs.ide.vbcs.optionslogger.logoptions.newlineforelse_csharpformattingoptions | TRUE
vs.ide.vbcs.optionslogger.logoptions.newlineforfinally_csharpformattingoptions | TRUE
vs.ide.vbcs.optionslogger.logoptions.newlineformembersinanonymoustypes_csharpformattingoptions | TRUE
vs.ide.vbcs.optionslogger.logoptions.newlineformembersinobjectinit_csharpformattingoptions | TRUE
vs.ide.vbcs.optionslogger.logoptions.newlinesforbracesinaccessors_csharpformattingoptions | TRUE
vs.ide.vbcs.optionslogger.logoptions.newlinesforbracesinanonymousmethods_csharpformattingoptions | TRUE
vs.ide.vbcs.optionslogger.logoptions.newlinesforbracesinanonymoustypes_csharpformattingoptions | TRUE
vs.ide.vbcs.optionslogger.logoptions.newlinesforbracesincontrolblocks_csharpformattingoptions | TRUE
vs.ide.vbcs.optionslogger.logoptions.newlinesforbracesinlambdaexpressionbody_csharpformattingoptions | TRUE
vs.ide.vbcs.optionslogger.logoptions.newlinesforbracesinmethods_csharpformattingoptions | TRUE
vs.ide.vbcs.optionslogger.logoptions.newlinesforbracesinobjectcollectionarrayinitializers_csharpformattingoptions | TRUE
vs.ide.vbcs.optionslogger.logoptions.newlinesforbracesinproperties_csharpformattingoptions | TRUE
vs.ide.vbcs.optionslogger.logoptions.newlinesforbracesintypes_csharpformattingoptions | TRUE
vs.ide.vbcs.optionslogger.logoptions.offerremoveunusedreferences_featureonoffoptions | TRUE
vs.ide.vbcs.optionslogger.logoptions.offerremoveunusedreferencesfeatureflag_featureonoffoptions | FALSE
vs.ide.vbcs.optionslogger.logoptions.oop64bit_internalfeatureonoffoptions | TRUE
vs.ide.vbcs.optionslogger.logoptions.oopcoreclrfeatureflag_internalfeatureonoffoptions | FALSE
vs.ide.vbcs.optionslogger.logoptions.oopservergcfeatureflag_internalfeatureonoffoptions | FALSE
vs.ide.vbcs.optionslogger.logoptions.openonfocus_stacktraceexploreroptions | FALSE
vs.ide.vbcs.optionslogger.logoptions.outputwindowloggerkey_loggeroptions | FALSE
vs.ide.vbcs.optionslogger.logoptions.partialloadmodefeatureflag_visualstudioworkspacestatusservice | TRUE
vs.ide.vbcs.optionslogger.logoptions.preferbraces_csharpcodestyleoptions | \<CodeStyleOption SerializationVersion="1"   Type="Boolean" Value="true"   DiagnosticSeverity="Hidden" />
vs.ide.vbcs.optionslogger.logoptions.preferconditionaldelegatecall_csharpcodestyleoptions | \<CodeStyleOption SerializationVersion="1"   Type="Boolean" Value="true"   DiagnosticSeverity="Info" />
vs.ide.vbcs.optionslogger.logoptions.preferdeconstructedvariabledeclaration_csharpcodestyleoptions | \<CodeStyleOption SerializationVersion="1"   Type="Boolean" Value="true"   DiagnosticSeverity="Info" />
vs.ide.vbcs.optionslogger.logoptions.preferexpressionbodiedaccessors_csharpcodestyleoptions | \<CodeStyleOption SerializationVersion="1"   Type="Boolean" Value="true"   DiagnosticSeverity="Hidden" />
vs.ide.vbcs.optionslogger.logoptions.preferexpressionbodiedconstructors_csharpcodestyleoptions | \<CodeStyleOption SerializationVersion="1"   Type="Boolean" Value="false"   DiagnosticSeverity="Hidden" />
vs.ide.vbcs.optionslogger.logoptions.preferexpressionbodiedindexers_csharpcodestyleoptions | \<CodeStyleOption SerializationVersion="1"   Type="Boolean" Value="true"   DiagnosticSeverity="Hidden" />
vs.ide.vbcs.optionslogger.logoptions.preferexpressionbodiedlambdas_csharpcodestyleoptions | \<CodeStyleOption SerializationVersion="1"   Type="Boolean" Value="true"   DiagnosticSeverity="Hidden" />
vs.ide.vbcs.optionslogger.logoptions.preferexpressionbodiedlocalfunctions_csharpcodestyleoptions | \<CodeStyleOption SerializationVersion="1"   Type="Boolean" Value="false"   DiagnosticSeverity="Hidden" />
vs.ide.vbcs.optionslogger.logoptions.preferexpressionbodiedmethods_csharpcodestyleoptions | \<CodeStyleOption SerializationVersion="1"   Type="Boolean" Value="false"   DiagnosticSeverity="Hidden" />
vs.ide.vbcs.optionslogger.logoptions.preferexpressionbodiedoperators_csharpcodestyleoptions | \<CodeStyleOption SerializationVersion="1"   Type="Boolean" Value="false"   DiagnosticSeverity="Hidden" />
vs.ide.vbcs.optionslogger.logoptions.preferexpressionbodiedproperties_csharpcodestyleoptions | \<CodeStyleOption SerializationVersion="1"   Type="Boolean" Value="true"   DiagnosticSeverity="Hidden" />
vs.ide.vbcs.optionslogger.logoptions.preferextendedpropertypattern_csharpcodestyleoptions | \<CodeStyleOption SerializationVersion="1"   Type="Boolean" Value="true"   DiagnosticSeverity="Info" />
vs.ide.vbcs.optionslogger.logoptions.preferindexoperator_csharpcodestyleoptions | \<CodeStyleOption SerializationVersion="1"   Type="Boolean" Value="true"   DiagnosticSeverity="Info" />
vs.ide.vbcs.optionslogger.logoptions.preferinlinedvariabledeclaration_csharpcodestyleoptions | \<CodeStyleOption SerializationVersion="1"   Type="Boolean" Value="true"   DiagnosticSeverity="Info" />
vs.ide.vbcs.optionslogger.logoptions.preferisnotexpression_visualbasiccodestyleoptions | \<CodeStyleOption SerializationVersion="1"   Type="Boolean" Value="true"   DiagnosticSeverity="Info" />
vs.ide.vbcs.optionslogger.logoptions.preferlocaloveranonymousfunction_csharpcodestyleoptions | \<CodeStyleOption SerializationVersion="1"   Type="Boolean" Value="true"   DiagnosticSeverity="Info" />
vs.ide.vbcs.optionslogger.logoptions.prefermethodgroupconversion_csharpcodestyleoptions | \<CodeStyleOption SerializationVersion="1"   Type="Boolean" Value="true"   DiagnosticSeverity="Hidden" />
vs.ide.vbcs.optionslogger.logoptions.prefernotpattern_csharpcodestyleoptions | \<CodeStyleOption SerializationVersion="1"   Type="Boolean" Value="true"   DiagnosticSeverity="Info" />
vs.ide.vbcs.optionslogger.logoptions.prefernullcheckovertypecheck_csharpcodestyleoptions | \<CodeStyleOption SerializationVersion="1"   Type="Boolean" Value="true"   DiagnosticSeverity="Info" />
vs.ide.vbcs.optionslogger.logoptions.preferparameternullchecking_csharpcodestyleoptions | \<CodeStyleOption SerializationVersion="1"   Type="Boolean" Value="true"   DiagnosticSeverity="Info" />
vs.ide.vbcs.optionslogger.logoptions.preferpatternmatching_csharpcodestyleoptions | \<CodeStyleOption SerializationVersion="1"   Type="Boolean" Value="true"   DiagnosticSeverity="Hidden" />
vs.ide.vbcs.optionslogger.logoptions.preferpatternmatchingoveraswithnullcheck_csharpcodestyleoptions | \<CodeStyleOption SerializationVersion="1"   Type="Boolean" Value="true"   DiagnosticSeverity="Info" />
vs.ide.vbcs.optionslogger.logoptions.preferpatternmatchingoveriswithcastcheck_csharpcodestyleoptions | \<CodeStyleOption SerializationVersion="1"   Type="Boolean" Value="true"   DiagnosticSeverity="Info" />
vs.ide.vbcs.optionslogger.logoptions.preferrangeoperator_csharpcodestyleoptions | \<CodeStyleOption SerializationVersion="1"   Type="Boolean" Value="true"   DiagnosticSeverity="Info" />
vs.ide.vbcs.optionslogger.logoptions.preferredmodifierorder_csharpcodestyleoptions | \<CodeStyleOption SerializationVersion="1"   Type="String"   Value="public,private,protected,internal,static,extern,new,virtual,abstract,sealed,override,readonly,unsafe,volatile,async"   DiagnosticSeverity="Hidden" />
vs.ide.vbcs.optionslogger.logoptions.preferredmodifierorder_visualbasiccodestyleoptions | \<CodeStyleOption SerializationVersion="1"   Type="String"   Value="Partial,Default,Private,Protected,Public,Friend,NotOverridable,Overridable,MustOverride,Overloads,Overrides,MustInherit,NotInheritable,Static,Shared,Shadows,ReadOnly,WriteOnly,Dim,Const,WithEvents,Widening,Narrowing,Custom,Async,Iterator"   DiagnosticSeverity="Hidden" />
vs.ide.vbcs.optionslogger.logoptions.preferredusingdirectiveplacement_csharpcodestyleoptions | \<CodeStyleOption SerializationVersion="1"   Type="Boolean" Value="true"   DiagnosticSeverity="Hidden" />
vs.ide.vbcs.optionslogger.logoptions.prefersimpledefaultexpression_csharpcodestyleoptions | \<CodeStyleOption SerializationVersion="1"   Type="Boolean" Value="true"   DiagnosticSeverity="Info" />
vs.ide.vbcs.optionslogger.logoptions.prefersimpleusingstatement_csharpcodestyleoptions | \<CodeStyleOption SerializationVersion="1"   Type="Boolean" Value="true"   DiagnosticSeverity="Info" />
vs.ide.vbcs.optionslogger.logoptions.prefersimplifiedobjectcreation_visualbasiccodestyleoptions | \<CodeStyleOption SerializationVersion="1"   Type="Boolean" Value="true"   DiagnosticSeverity="Info" />
vs.ide.vbcs.optionslogger.logoptions.preferstaticlocalfunction_csharpcodestyleoptions | \<CodeStyleOption SerializationVersion="1"   Type="Boolean" Value="true"   DiagnosticSeverity="Info" />
vs.ide.vbcs.optionslogger.logoptions.preferswitchexpression_csharpcodestyleoptions | \<CodeStyleOption SerializationVersion="1"   Type="Boolean" Value="true"   DiagnosticSeverity="Info" />
vs.ide.vbcs.optionslogger.logoptions.preferthrowexpression_csharpcodestyleoptions | \<CodeStyleOption SerializationVersion="1"   Type="Boolean" Value="true"   DiagnosticSeverity="Info" />
vs.ide.vbcs.optionslogger.logoptions.prefertoplevelstatements_csharpcodestyleoptions | \<CodeStyleOption SerializationVersion="1"   Type="Boolean" Value="true"   DiagnosticSeverity="Hidden" />
vs.ide.vbcs.optionslogger.logoptions.prefertupleswap_csharpcodestyleoptions | \<CodeStyleOption SerializationVersion="1"   Type="Boolean" Value="true"   DiagnosticSeverity="Info" />
vs.ide.vbcs.optionslogger.logoptions.preferutf8stringliterals_csharpcodestyleoptions | \<CodeStyleOption SerializationVersion="1"   Type="Boolean" Value="true"   DiagnosticSeverity="Info" />
vs.ide.vbcs.optionslogger.logoptions.removerecommendationlimit_internalfeatureonoffoptions | FALSE
vs.ide.vbcs.optionslogger.logoptions.removeunnecessarysuppressionexclusions_codestyleoptions |  
vs.ide.vbcs.optionslogger.logoptions.renametracking_internalfeatureonoffoptions | TRUE
vs.ide.vbcs.optionslogger.logoptions.resharperstatus_keybindingresetoptions | NotInstalledOrDisabled
vs.ide.vbcs.optionslogger.logoptions.semanticcolorizer_internalfeatureonoffoptions | TRUE
vs.ide.vbcs.optionslogger.logoptions.showdebuginfo_internalfeatureonoffoptions | FALSE
vs.ide.vbcs.optionslogger.logoptions.skipanalyzersforimplicitlytriggeredbuilds_featureonoffoptions | TRUE
vs.ide.vbcs.optionslogger.logoptions.smartindenter_internalfeatureonoffoptions | TRUE
vs.ide.vbcs.optionslogger.logoptions.snippets_internalfeatureonoffoptions | TRUE
vs.ide.vbcs.optionslogger.logoptions.solutionchecksummonitorbackofftimespaninms_internalfeatureonoffoptions | 1000
vs.ide.vbcs.optionslogger.logoptions.spaceaftercast_csharpformattingoptions | FALSE
vs.ide.vbcs.optionslogger.logoptions.spaceaftercoloninbasetypedeclaration_csharpformattingoptions | TRUE
vs.ide.vbcs.optionslogger.logoptions.spaceaftercomma_csharpformattingoptions | TRUE
vs.ide.vbcs.optionslogger.logoptions.spaceaftercontrolflowstatementkeyword_csharpformattingoptions | TRUE
vs.ide.vbcs.optionslogger.logoptions.spaceafterdot_csharpformattingoptions | FALSE
vs.ide.vbcs.optionslogger.logoptions.spaceaftermethodcallname_csharpformattingoptions | FALSE
vs.ide.vbcs.optionslogger.logoptions.spaceaftersemicolonsinforstatement_csharpformattingoptions | TRUE
vs.ide.vbcs.optionslogger.logoptions.spacebeforecoloninbasetypedeclaration_csharpformattingoptions | TRUE
vs.ide.vbcs.optionslogger.logoptions.spacebeforecomma_csharpformattingoptions | FALSE
vs.ide.vbcs.optionslogger.logoptions.spacebeforedot_csharpformattingoptions | FALSE
vs.ide.vbcs.optionslogger.logoptions.spacebeforeopensquarebracket_csharpformattingoptions | FALSE
vs.ide.vbcs.optionslogger.logoptions.spacebeforesemicolonsinforstatement_csharpformattingoptions | FALSE
vs.ide.vbcs.optionslogger.logoptions.spacebetweenemptymethodcallparentheses_csharpformattingoptions | FALSE
vs.ide.vbcs.optionslogger.logoptions.spacebetweenemptymethoddeclarationparentheses_csharpformattingoptions | FALSE
vs.ide.vbcs.optionslogger.logoptions.spacebetweenemptysquarebrackets_csharpformattingoptions | FALSE
vs.ide.vbcs.optionslogger.logoptions.spacesignorearoundvariabledeclaration_csharpformattingoptions | FALSE
vs.ide.vbcs.optionslogger.logoptions.spacewithincastparentheses_csharpformattingoptions | FALSE
vs.ide.vbcs.optionslogger.logoptions.spacewithinexpressionparentheses_csharpformattingoptions | FALSE
vs.ide.vbcs.optionslogger.logoptions.spacewithinmethodcallparentheses_csharpformattingoptions | FALSE
vs.ide.vbcs.optionslogger.logoptions.spacewithinmethoddeclarationparenthesis_csharpformattingoptions | FALSE
vs.ide.vbcs.optionslogger.logoptions.spacewithinotherparentheses_csharpformattingoptions | FALSE
vs.ide.vbcs.optionslogger.logoptions.spacewithinsquarebrackets_csharpformattingoptions | FALSE
vs.ide.vbcs.optionslogger.logoptions.spacingaftermethoddeclarationname_csharpformattingoptions | FALSE
vs.ide.vbcs.optionslogger.logoptions.spacingaroundbinaryoperator_csharpformattingoptions | Single
vs.ide.vbcs.optionslogger.logoptions.squiggles_internalfeatureonoffoptions | TRUE
vs.ide.vbcs.optionslogger.logoptions.syntacticcolorizer_internalfeatureonoffoptions | TRUE
vs.ide.vbcs.optionslogger.logoptions.tagger_editorcomponentonoffoptions | TRUE
vs.ide.vbcs.optionslogger.logoptions.traceloggerkey_loggeroptions | FALSE
vs.ide.vbcs.optionslogger.logoptions.unusedvalueassignment_csharpcodestyleoptions | \<CodeStyleOption SerializationVersion="1"   Type="Int32" Value="2"   DiagnosticSeverity="Info" />
vs.ide.vbcs.optionslogger.logoptions.unusedvalueassignment_visualbasiccodestyleoptions | \<CodeStyleOption SerializationVersion="1"   Type="Boolean" Value="true"   DiagnosticSeverity="Info" />
vs.ide.vbcs.optionslogger.logoptions.unusedvalueexpressionstatement_csharpcodestyleoptions | \<CodeStyleOption SerializationVersion="1"   Type="Int32" Value="2"   DiagnosticSeverity="Hidden" />
vs.ide.vbcs.optionslogger.logoptions.unusedvalueexpressionstatement_visualbasiccodestyleoptions | \<CodeStyleOption SerializationVersion="1"   Type="Boolean" Value="true"   DiagnosticSeverity="Hidden" />
vs.ide.vbcs.optionslogger.logoptions.varelsewhere_csharpcodestyleoptions | \<CodeStyleOption SerializationVersion="1"   Type="Boolean" Value="false"   DiagnosticSeverity="Hidden" />
vs.ide.vbcs.optionslogger.logoptions.varforbuiltintypes_csharpcodestyleoptions | \<CodeStyleOption SerializationVersion="1"   Type="Boolean" Value="false"   DiagnosticSeverity="Hidden" />
vs.ide.vbcs.optionslogger.logoptions.varwhentypeisapparent_csharpcodestyleoptions | \<CodeStyleOption SerializationVersion="1"   Type="Boolean" Value="false"   DiagnosticSeverity="Hidden" />
vs.ide.vbcs.optionslogger.logoptions.wrappingkeepstatementsonsingleline_csharpformattingoptions | TRUE
vs.ide.vbcs.optionslogger.logoptions.wrappingpreservesingleline_csharpformattingoptions | TRUE



</body>

</html>

**Comment by ryzngard** [comment]
> Should we log these under one property, so we don't need to classify each new option added for GDPR?

I was thinking about that, following up with Mika today on what would be best here. There could be cases where GDPR tagging would actually benefit us (in case we have options with custom user input like file header)

**Comment by davidwengier** [comment]
@vatsalyaagrawal happened to mention in standup today that there were hundreds of telemetry events to classify... related? 😁

**Comment by ryzngard** [comment]
> @vatsalyaagrawal happened to mention in standup today that there were hundreds of telemetry events to classify... related? 😁

If only there was a way to know that we're running a debug build and not report telemetry.... 👀

**Comment by dibarbet** [comment]
> If only there was a way to know that we're running a debug build and not report telemetry.... 👀

Sometimes you want it - I always want it in the telemetry monitor (to verify my events are getting sent as expected).  I sometimes like having it actually show up in the gdpr portal so I don't have to wait until it gets merged into VS to classify it.

**Comment by davidwengier** [comment]
You make good points @dibarbet but I've also had to classify my own typos, so there are definitely downsides :)

File: src/VisualStudio/Core/Def/OptionsLogger.cs [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        }
+
+        public void Dispose()
+        {
+            _solutionEventMonitor.SolutionClosed -= OnSolutionClosed;
+        }
+
+        internal static void ReportTelemetry(Workspace workspace, CancellationToken cancellationToken)
+        {
+            var optionsService = workspace.Services.GetRequiredService<IOptionService>();
```
Comment (sharwell) [inline src/VisualStudio/Core/Def/OptionsLogger.cs:44]: Can we make this return without doing any work when telemetry is disabled?

**Comment by mikadumont** [comment]
Design Meeting Notes: We are not going to do .editorconfig options for now just global options. We will work internally with the telemetry team on this.

**Comment by ryzngard** [comment]
Closing an old PR. Guessing this is not needed :)

**Pull request closed without merging**